src/pages/keyword-generator/KeywordGenerator.vue

```diff
@@ -25,154 +27,82 @@ const toggleNgramCheckbox = (ngramSize) => {
-           @click="toggleNgramCheckbox(ngramSize)">
-        <input type="checkbox" :id="`ngram-${ngramSize}`" :value="ngramSize" v-model="selectedNgramSizes[ngramSize-1]"
-               @click.stop/>
-        <label :for="`ngram-${ngramSize}`" @click="toggleNgramCheckbox(ngramSize)">{{ ngramSize }}-gram</label>
+  <div class="w-5/6 mx-auto">
+    <div class="md:w-2/3 mx-auto">
+      <h3 class="font-sans font-bold text-center my-10 text-3xl">Keyword Generator</h3>
+      <div class="flex max-sm:flex-col md:flex-row">
+        <ma-input
+            type="textarea"
+            placeholder="Enter the text to generate keywords from"
+            rows=5
+            size="large"
+            class="basis-1/2"
+            v-model:value="keywordText"
+            @change="uncheckInvisibleCheckboxes()"
+        >
+        </ma-input>
+        <div class="grid grid-cols-4 gap-2 px-3">
+          <ma-checkbox v-for="ngramSize in Math.min(MAX_VALUE_OF_NGRAM_SIZE, cleanKeywords.length)"
+                       v-bind:key="ngramSize" @click="toggleNgramCheckbox(ngramSize)"> {{ ngramSize }}-gram
+          </ma-checkbox>
+        </div>
       </div>
     </div>
-    <div class="ma-ngrams-wrapper">
-      <div v-for="ngramData in ngramsList" v-bind:key="ngramData" class="ma-ngrams-column">
-        <div class="ma-ngrams-column-title">
-          {{ ngramData.size }}-grams
-        </div>
-        <div v-for="ngram in ngramData.grams.value" v-bind:key="ngram" class="ma-ngrams-ngram">
-          <ma-badge v-for="ngramWord in ngram.split(' ')" v-bind:key="ngramWord" variant="dark" size="large"
-                   type="primary" class="ma-ngrams-badge">
-            <div class="ma-ngrams-ngram-text">
-              {{ ngramWord }}
-            </div>
+    <div class="md:w-2/3 mx-auto my-3">
+      <ma-button type="primary" variant="dark" class="" icon="add" @click="generateNGrams()">Generate Ngrams</ma-button>
+    </div>
+    <div class="mx-auto mt-5">
+      <div class="flex flex-row flex-wrap justify-center my-3">
+        <div v-for="ngramData in ngramsList" v-bind:key="ngramData" class="shrink mx-2">
+          <ma-badge variant="dark" size="large" type="primary" class="py-1">
+            <h1 class="font-bold text-lg mx-auto">
+              {{ ngramData.size }}-grams
+            </h1>
           </ma-badge>
+          <div v-for="ngram in ngramData.grams" v-bind:key="ngram"
+               class="group my-1 p-1 rounded-full flex flex-row flex-wrap justify-center hover:bg-gray-800">
+            <ma-badge v-for="ngramWord in ngram" v-bind:key="ngramWord" variant="dark" size="large"
+                      type="primary" class="transition text-center mr-0.5 group-hover:bg-gray-100 group-hover:text-gray-800">
+              {{ ngramWord }}
+            </ma-badge>
+          </div>
         </div>
       </div>
     </div>
   </div>
```

Comment:
Can you use @apply directive as style in this component, see this [doc](https://tailwindcss.com/docs/functions-and-directives#apply)